test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMultipartMixedBatchWriterTests.cs

```diff
@@ -0,0 +1,797 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataMultipartMixedBatchWriterTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.OData.Edm;
+using Microsoft.OData.MultipartMixed;
+using Xunit;
+
+namespace Microsoft.OData.Tests
+{
+    public class ODataMultipartMixedBatchWriterTests
+    {
+        private const string ServiceUri = "http://tempuri.org";
+        private const string batchBoundary = "batch_aed653ab";
+        private ODataMediaType mediaType;
+        private MemoryStream stream;
+        private ODataMessageWriterSettings settings;
+
+        private EdmModel model;
+        private EdmEntityType orderEntityType;
+        private EdmEntityType customerEntityType;
+        private EdmEntitySet orderEntitySet;
+        private EdmEntitySet customerEntitySet;
+
+        public ODataMultipartMixedBatchWriterTests()
+        {
+            this.InitializeEdmModel();
+            this.stream = new MemoryStream();
+            this.settings = new ODataMessageWriterSettings
+            {
+                ODataUri = new ODataUri { ServiceRoot = new Uri(ServiceUri) }
+            };
+            this.settings.SetServiceDocumentUri(new Uri(ServiceUri));
+            this.mediaType = new ODataMediaType(
+                "Multipart",
+                "Mixed",
+                new KeyValuePair<string, string>[] { new KeyValuePair<string, string>("boundary", batchBoundary) });
+        }
+
+        [Fact]
+        public async Task WriteMultipartMixedBatchAsync()
+        {
+            var result = await SetupMultipartMixedBatchWriterAndRunTestAsync(
+                async (multipartMixedBatchWriter) =>
+                {
+                    await multipartMixedBatchWriter.WriteStartBatchAsync();
+                    await multipartMixedBatchWriter.WriteEndBatchAsync();
+                });
+
+            Assert.Equal("--batch_aed653ab--\r\n", result);

```

Comment:
WriteStartBatch should write `--batch_aed653ab`
WriteEndBatch should write `--batch_aed653ab--`
So what should be the result?

Reply:
@KenitoInc In the existing implementation, writing of the batch start boundary is triggered by either of the following 3 events
- Creation of a operation request message
- Creation of a operation response message
- Start of a changeset

This test is more for an experimental rather than practical scenario and I'd also agree with you that the logical outcome should be `--batch_aed653ab\r\n--batch_aed653ab--\r\n`.
In addition the fact that calling only `WriteStartBatch` and `WriteEndBatch` serves no practical purposes in batch writing scenario, attempting to rectifying the logic such that the start boundary is written when only the two methods are called on multipart-mixed batch writer would be considered a breaking change since `--batch_aed653ab--\r\n` is the outcome for the already existing synchronous workflow.
I have however modified the test to trigger `CreateOperationRequestMessageAsync` to reflect a more logical outcome